### PR TITLE
Urlencode and decode session values in headers

### PR DESCRIPTION
**Pull request by chyzzqo2**

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -358,6 +358,49 @@ def assert_headers(headers):
...
+        assert headers[constants.HEADER_SOURCE] == source
+        assert headers[constants.HEADER_USER] == user
+        assert headers[constants.HEADER_SESSION] == "a=1,b=2,c=more%3Dv1%2Cv2"
+        assert len(headers.keys()) == 6
+
+    req.post("URL")
+    assert_headers(post_recorder.kwargs["headers"])
+
+    req.get("URL")
+    assert_headers(get_recorder.kwargs["headers"])
```
Comment (ebyhr) [inline tests/test_client.py:401]: Let's focus on testing `session_properties` parameter. e.g.
```python
def test_request_session_headers(monkeypatch):
    post_recorder = ArgumentsRecorder()
    monkeypatch.setattr(TrinoRequest.http.Session, "post", post_recorder)

    get_recorder = ArgumentsRecorder()
    monkeypatch.setattr(TrinoRequest.http.Session, "get", get_recorder)

    req = TrinoRequest(
        host="coordinator",
        port=8080,
        user="test_user",
        session_properties={
            "a": "1",
            "b": "2",
            "c": "more=v1,v2"
        }
    )

    def assert_headers(headers):
        assert headers[constants.HEADER_SESSION] == "a=1,b=2,c=more%3Dv1%2Cv2"

    req.post("URL")
    assert_headers(post_recorder.kwargs["headers"])

    req.get("URL")
    assert_headers(get_recorder.kwargs["headers"])
```

File: tests/test_client.py [hunk 2]
```diff
@@ -358,6 +358,49 @@ def assert_headers(headers):
     assert_headers(get_recorder.kwargs["headers"])
 
 
+def test_request_session_headers(monkeypatch):
```
Comment (ebyhr) [inline tests/test_client.py:361]: nit: Rename to `test_request_session_properties_headers`

File: tests/test_client.py (hunk 1 above)
Comment (chyzzqo2) [reply tests/test_client.py:386]: updated

**Comment by chyzzqo2** [comment]
Not sure why it's still not picking up the CLA, but made the other requested changes.

**Review by hashhar — APPROVED** [review]

**Comment by ebyhr** [comment]
@chyzzqo2 It seems the account of the commit `@cHYzZQo` is different from CLA. Could you fix it and push forcibly?

**Comment by chyzzqo2** [comment]
This ready to go in? Anything else needed from me?

**Pull request merged**